### PR TITLE
add header_icons_component tests, allow for null/missing-icons in res…

### LINKED ISSUE #1412: fix bug for specific header icons
Fix potential bug for resource type where the resource type doesn't have an icon or the value of resource_type is null

### PR DESCRIPTION
**Pull request by dnoneill**
…ource type

closes #1412

### CODE REVIEW

File: spec/components/geoblacklight/header_icons_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  context "when there is a resource type without icon" do
+    let(:resource_type_list) { ["Index Maps"] }
+    it "renders 3 icons" do
+      expect(rendered).to have_css(".blacklight-icon-datasets")
+      expect(rendered).to have_css(".blacklight-icon-stanford")
+      expect(rendered).to have_css(".blacklight-icon-public")
+    end
+  end
+
+  context "when there is a resource type is nil" do
```
Comment (thatbudakguy) [inline spec/components/geoblacklight/header_icons_component_spec.rb:38]:
```suggestion
  context "when the resource type is nil" do
```

**Review by thatbudakguy — APPROVED** [review]

**Review by thatbudakguy — APPROVED** [review]
